cmd/profiles.go

```diff
@@ -0,0 +1,165 @@
+	"os"
+	"strings"
+)
+
+func init() {
+	legalActions := map[string]interface{}{
+		"add":     add,
+		"create":  add,
+		"list":    list,
+		"ls":      list,
+		"show":    show,
+		"use":     use,
+		"edit":    edit,
+		"update":  edit,
+		"remove":  remove,
+		"delete":  remove,
+		"which":   current,
+		"current": current,
+	}
+	rootCmd.AddCommand(&cobra.Command{
+		Use:   "init",
+		Short: "Creates a default OneLogin Profile",
+		Long:  "Creates and activates the first OneLogin Profile",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			configFile, err := os.OpenFile(viper.ConfigFileUsed(), os.O_RDWR, 0600)
+			if err != nil {
+				configFile.Close()
+				log.Fatalln("Unable to open profiles file", err)
+			}
+			profileService := profiles.ProfileService{
+				Repository:  profiles.FileRepository{StorageMedia: configFile},
+				InputReader: os.Stdin,
+			}
+			if len(profileService.Index()) > 0 {
+				configFile.Close()
+				log.Fatalln("Profiles already set up!")
+			}
+			configFile.Close()
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			configFile, err := os.OpenFile(viper.ConfigFileUsed(), os.O_RDWR, 0600)
+			if err != nil {
+				configFile.Close()
+				log.Fatalln("Unable to open profiles file", err)
+			}
+			profileService := profiles.ProfileService{
+				Repository:  profiles.FileRepository{StorageMedia: configFile},
+				InputReader: os.Stdin,
+			}
+			profileService.Create("default")
+			configFile.Close()
+		},
+	})
+	rootCmd.AddCommand(&cobra.Command{
+		Use:   "profiles",
+		Short: "Manage account settings for the CLI",
+		Long: `Maintains a listing of accounts used by the CLI in a home/.onelogin/profiles file
+		and facilitates creating, changing, deleting, indexing, and using known configurations.
+		You are of course, free to go and edit the profiles file yourself and use this as a way to quickly switch out your environment.
+		Available Actions:
+			use             [name - required] => marks this profile as active so the CLI will use it's credentials in all requests to OneLogin
+			show            [name - required] => shows information about the profile
+			edit   (update) [name - required] => edits selected account information
+			remove (delete) [name - required] => removes selected account
+			add    (create) [name - required] => adds account to manage
+			list   (ls)                       => lists managed accounts that can be used. if name given, lists information about that profile
+			which  (current)                  => returns current active profile`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				log.Fatalf("Must specify action to perform!")
+			}
+			action := strings.ToLower(args[0])
+			if legalActions[action] == nil {
+				log.Fatalf("Illegal Action!")
+			}
+			if (action == "show" || action == "add" || action == "use" || action == "edit" || action == "update" || action == "remove" || action == "delete") && len(args) < 2 {
```

Comment:
Since these are static strings, it might be nice to have a function that checks for actions that require a profile name, using a switch:
```Golang
switch category {
case
  "show", "add", "use", "edit", "update", "remove", "delete"
  return true
}
return false
```

Reply:
Good call! I figured there should be a cleaner way to do this.